lib/rucio/web/rest/flaskapi/v1/common.py

```diff
@@ -192,7 +191,43 @@ def parse_scope_name(scope_name, vo):
     return scope_name.group(1, 2)
 
 
-def try_stream(generator, content_type=None) -> "flask.Response":
+def _cut_json_stream(generator: typing.Iterator[typing.AnyStr]) -> typing.Iterator[typing.AnyStr]:
+    """
+    Generator for cutting off JSON document endings until
+    the next element from the generator arrived.
+    """
+    last_document_end = None
+    for part in generator:
+        is_json_doc = False
+        document_end_index = -1
+        if isinstance(part, bytes):
+            json_document_end = b'}'
+            is_json_doc = part.rstrip().endswith(json_document_end)
+            if is_json_doc:
+                document_end_index = part.rfind(json_document_end)
+        elif isinstance(part, str):
+            json_document_end = '}'
+            is_json_doc = part.rstrip().endswith(json_document_end)
+            if is_json_doc:
+                document_end_index = part.rfind(json_document_end)
+
+        if is_json_doc:
+            if last_document_end:
+                yield last_document_end + part[:document_end_index]
+            else:
+                yield part[:document_end_index]
+            last_document_end = part[document_end_index:]
+        else:
+            if last_document_end:
+                yield last_document_end + part
+            else:
+                yield part
+
+    if last_document_end:
+        yield last_document_end
```

Comment:
This seems like a suitable solution, but why are we doing it this complicated?
Wouldn't it be enough to return an incorrect string once the error is thrown? Something like:
```python
def _cut_json_stream(generator: typing.Iterator[typing.AnyStr]) -> typing.Iterator[typing.Union[str, typing.AnyStr]]:
    """
    Generator for cutting off JSON document endings until
    the next element from the generator arrived.
    """
    try:
        yield from generator
    except Exception as e:
        yield str(e) + "-"
        raise e
```

The coresponding test would still recieve the two elements, but then an error:
```python
def test_try_stream_error_after_two_items(flask_application):
    first_object = {'some': 'object'}
    second_object = {'another': 'object'}

    def generate_with_error():
        yield json.dumps(first_object) + '\n'
        yield json.dumps(second_object) + '\n'
        raise RuntimeError('Error for testing')

    with flask_application.app_context():
        with flask_application.test_request_context('/test_try_stream_error_after_two_items'):
            response_bytes: typing.List[bytes] = []
            with pytest.raises(RuntimeError, match='Error for testing'):
                _try_stream_to_bytes(flask_application, response_bytes, generate_with_error())
            response_docs = _cut_at_newlines(response_bytes)
            assert len(response_docs) == 3

            first_object_response = json.loads(response_docs[0])
            assert first_object_response == first_object
            first_object_response = json.loads(response_docs[1])
            assert first_object_response == second_object
            with pytest.raises(json.JSONDecodeError):
                print("Cropped JSON response from third object:", repr(response_docs[2]))
                json.loads(response_docs[2])
```

And the test output:
```
b'{"some": "object"}\n'
b'{"another": "object"}\n'
b'Error for testing-'
Cropped JSON response from third object: b'Error for testing-'
```

Reply:
Simply because we do not know who is closing the connection. The method I implemented also works with any reverse proxies or underlying network issues closing the HTTP connection.

Reply:
Ah ok, I understand now. Thanks! :sweat_smile: